src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -239,21 +260,21 @@ private void enterPosition(Spread spread) {
             activePosition.setExitTarget(exitTarget);
             activePosition.setEntryBalance(totalBalance);
             activePosition.getLongTrade().setExchange(spread.getLongExchange());
-            activePosition.getLongTrade().setVolume(longVolume);
+            activePosition.getLongTrade().setVolume(tradeVolume.getLongVolume());
             activePosition.getLongTrade().setEntry(longLimitPrice);
             activePosition.getShortTrade().setExchange(spread.getShortExchange());
-            activePosition.getShortTrade().setVolume(shortVolume);
+            activePosition.getShortTrade().setVolume(tradeVolume.getShortVolume());
             activePosition.getShortTrade().setEntry(shortLimitPrice);
 
             executeOrderPair(
                 spread.getLongExchange(), spread.getShortExchange(),
                 spread.getCurrencyPair(),
                 longLimitPrice, shortLimitPrice,
-                longVolumeWithFeesAndAdjustedStep, shortVolumeWithFeesAndAdjustedStep,
+                tradeVolume.getLongOrderVolume(), tradeVolume.getShortOrderVolume(),
                 true);
 
-            notificationService.sendEmailNotificationBodyForEntryTrade(spread, exitTarget, longVolume,
-                longLimitPrice, shortVolume, shortLimitPrice);
+            notificationService.sendEmailNotificationBodyForEntryTrade(spread, exitTarget, tradeVolume.getLongVolume(),
```

Comment:
Can you pass in here the minimum profit and the market neutrality value and then format the email and instant notification (telegram message) to include them?

First you need to merge master into this branch otherwise you won't have this new instant message code.